test/test_va_api_fixture.cpp

```diff
@@ -93,21 +95,61 @@ static VADisplay getDrmDisplay(int &fd)
 {
     typedef std::vector<std::string> DevicePaths;
     typedef DevicePaths::const_iterator DevicePathsIterator;
+    drmDevicePtr *devices;
+    int ret, max_devices;
 
-    static DevicePaths paths({"/dev/dri/renderD128", "/dev/dri/card0"});
-
-    const DevicePathsIterator endIt(paths.end());
-    for (DevicePathsIterator it(paths.begin()); it != endIt; ++it) {
-        fd = open(it->c_str(), O_RDWR);
-        if (fd < 0)
-            continue;
-        VADisplay disp = vaGetDisplayDRM(fd);
+    max_devices = drmGetDevices2(0, NULL, 0);
+    EXPECT_TRUE(max_devices > 0);
+    if (max_devices <= 0)
+        return NULL;
 
-        if (disp)
-            return disp;
+    devices = static_cast<drmDevicePtr*>(calloc(max_devices, sizeof(*devices)));
+    EXPECT_TRUE(devices);
+    if (!devices)
+        return NULL;
 
-        close(fd);
+    ret = drmGetDevices2(0, devices, max_devices);
+    EXPECT_TRUE(ret >= 0);
+    if (ret < 0) {
+        free(devices);
+        return NULL;
+    }
+    max_devices = ret;
+
+    for (int i = 0; i < max_devices; i++) {
+        for (int j = 0; j < DRM_NODE_MAX; j++) {
```

Comment:
The `j` (aka node type) should be decreasing, since as-is this is picks the `card` (primary) node before the render one.

Reply:
done